toolchain/semantics/semantics_handle_index.cpp

```diff
@@ -5,17 +5,56 @@
 #include <utility>
 
 #include "toolchain/semantics/semantics_context.h"
+#include "toolchain/semantics/semantics_node.h"
+#include "toolchain/semantics/semantics_node_kind.h"
 
 namespace Carbon {
 
 auto SemanticsHandleIndexExpressionStart(SemanticsContext& context,
                                          ParseTree::Node parse_node) -> bool {
-  return context.TODO(parse_node, "HandleIndexExpressionStart");
+  context.node_stack().Push(parse_node, context.node_stack().PopExpression());
+  return true;
 }
 
 auto SemanticsHandleIndexExpression(SemanticsContext& context,
                                     ParseTree::Node parse_node) -> bool {
-  return context.TODO(parse_node, "HandleIndexExpression");
+  auto ind_node = context.node_stack().PopExpression();
+  auto [ind_expr_parse_node, name_node_id] =
```

Comment:
It looks like ind_expr_parse_node is unused, so you shouldn't need to fetch it. Note the "WithParseNode" is optional.